modulemd/include/modulemd-2.0/modulemd-module-stream-v1.h

```diff
@@ -257,6 +257,23 @@ const gchar *
 modulemd_module_stream_v1_get_tracker (ModulemdModuleStreamV1 *self);
 
 
+/**
+ * modulemd_module_stream_v1_get_obsoletes_resolved:
+ * @self: (in): This #ModulemdModuleStreamV1 object.
+ *
+ * Returns: (transfer none): The #ModulemdObsoletes information associated with
+ * this object. If the associated obsoletes has reset attribute set, this
+ * function doesn't return it. From outside obsoletes with reset looks like
+ * there is no obsoletes set for this stream. Every obsoletes (even with reset) can
+ * be accessed from the streams module.
+ *
+ * Since: 2.10
+ */
+ModulemdObsoletes *
+modulemd_module_stream_v1_get_obsoletes_resolved (
+  ModulemdModuleStreamV1 *self);
+
+
```

Comment:
I recommend *against* implementing this for ModuleStreamV1 documents. We do not have any of these in the repos at this point. (All RHEL 8.0 module streams are V2 and Fedora no longer has any supported releases with V1 content in them). A better approach would be to document that the use of `modulemd_module_index_add_obsoletes()` will automatically update the stream object to at least V2 as a side-effect. As it is, the ModuleIndex always upgrades every object inside it to match the highest Stream version it sees, so there is precedent for this.

Let's try to avoid enhancing ModuleStreamV1 ever again, please.

Reply:
oh, of course, I can drop the commit and add the description.